test/Constraints/opened_existentials.swift

```diff
@@ -591,3 +591,19 @@ protocol PP3 {
     associatedtype A
 }
 
+protocol PP4 {
+    func check() -> Bool
+}
+
+do {
+  func test<T>(env: T) where T: PP4 {}
+
+  func test(env: PP4? = nil) {
+    guard let env else {
```

Comment:
Is the guard let essential to the test case?

Reply:
I should note that today existential opening doesn't work for stuff like `env ?? <<default>>` because it currently happens in `simplifyApplicableFnConstraint` and at that point the argument won't be resolved. This is a bug we need to fix.

Reply:
Yep!